File: c week 2/assignment5.c
#include <stdio.h>

#define MAX 100

// Function to check if matrix is sparse
int isSparse(int array[MAX][MAX], int m, int n) {
    int counter = 0;

    // Count the number of zeros in the matrix
    for (int i = 0; i < m; i++) {
        for (int j = 0; j < n; j++) {
            if (array[i][j] == 0) {
                counter++;
            }
        }
    }

    // If number of zeros > half of total elements
    return (counter > (m * n) / 2);
}

int main() {
    int array[MAX][MAX] = {
        {1, 0, 3},
        {0, 0, 4},
        {6, 0, 0}
    };

    int m = 3, n = 3;

    if (isSparse(array, m, n)) {
        printf("Yes\n");
    } else {
        printf("No\n");
    }

    return 0;
}

File: python week 2/assignment5.py
# Python 3 code to check
# if a matrix is
# sparse.

MAX = 100
 
def isSparse(array,m, n) :
    
    counter = 0
 
    # Count number of zeros
    # in the matrix
    for i in range(0,m) :
        for j in range(0,n) :
            if (array[i][j] == 0) :
                counter = counter + 1
 
    return (counter > 
            ((m * n) // 2))

# Driver Function
array = [ [ 1, 0, 3 ],
          [ 0, 0, 4 ],
          [ 6, 0, 0 ] ]
m = 3
n = 3

if (isSparse(array, m, n)) :
    print("Yes")
else :
    print("No")
    
    
# this code is contributed by
# Nikita tiwari